crates/ra_flycheck/src/lib.rs

```diff
@@ -216,27 +214,34 @@ impl FlycheckThread {
         self.message_recv = never();
         self.check_process = None;
 
-        let mut args: Vec<String> = vec![
-            self.options.command.clone(),
-            "--workspace".to_string(),
-            "--message-format=json".to_string(),
-            "--manifest-path".to_string(),
-            format!("{}/Cargo.toml", self.workspace_root.display()),
-        ];
-        if self.options.all_targets {
-            args.push("--all-targets".to_string());
-        }
-        args.extend(self.options.args.iter().cloned());
+        let mut cmd = match &self.config {
+            FlycheckConfig::CargoCommand { command, all_targets, extra_args } => {
+                let mut cmd = Command::new(cargo_binary());
+                cmd.arg(command);
+                cmd.args(&["--workspace", "--message-format=json", "--manifest-path"]);
+                cmd.arg(self.workspace_root.join("Cargo.toml"));
+                if *all_targets {
+                    cmd.arg("--all-targets");
+                }
+                cmd.args(extra_args);
+                cmd
+            }
+            FlycheckConfig::CustomCommand { command, args } => {
+                let mut cmd = Command::new(command);
+                cmd.args(args);
```

Comment:
I think it would be nice to support `--manifest-path` here as well. Maybe some special arg like `MANIFEST_PATH` that will be replaced here with the manifest path?

Reply:
Why the user can't add `--manifest-path=/path/to/Cargo.toml` themselves?

Reply:
This would require some dynamic configuration, while the other way would support having a static configuration. It would be solvable, but I think it requires more work for the user than doing it here.